libs/json/structs.go

```diff
@@ -0,0 +1,71 @@
+package json
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"unicode"
+
+	lru "github.com/hashicorp/golang-lru"
+)
+
+// structInfoCache caches struct info.
+var structInfoCache *lru.Cache
```

Comment:
I'm not really sure if we need this. It caches struct metadata, specifically decoded JSON tags (e.g. `json:"name,omitempy"`), during struct encoding. However, since there is a finite number of structs in the codebase, maybe we should just cache all of them.

Reply:
Interesting idea. The whole cache would probably just take a few kb in mem anyways? If that's the case I think it is reasonable to just cache everything 👍

Reply:
Yeah, I think that makes sense. It'll probably be on the order of MBs, not KBs, but nothing major. I'll do some quick benchmarks to check, and throw out the LRU.